core/errors/infer.cc

```diff
@@ -9,17 +9,17 @@ ErrorClass errorClassForUntyped(const GlobalState &gs, FileRef file) {
     }
 
     auto isOpenInClient = file.data(gs).isOpenInClient();
-    if (!isOpenInClient) {
+    if (gs.printingFileTable) {
         // Note: this metric, despite being a prod metric, will not get reported in the normal way
         // to the metrics file, the web trace file, nor statsd. We call getAndClearHistogram BEFORE
         // calling getAndClearThreadCounters, which means that the metric will have been deleted
-        // before reporting to SignalFX.
+        // before reporting to SignalFX, and we don't even compute this if we are not running that
+        // code path (i.e. printing in realmain), because:
```

Comment:
I agree that we no longer compute this if we are not printing in realmain.  I was going to say that `pipeline::typecheck` still clears counters, but it clears them on _child_ threads, not the main thread, and it is the latter that is important here.  So I think this is OK (though you may want to double-check my logic).